src/lib/dnssd/Advertiser.h

```diff
@@ -124,7 +131,7 @@ class BaseAdvertisingParams
     uint8_t mMacStorage[kMaxMacSize] = {};
     size_t mMacLength                = 0;
     std::optional<ReliableMessageProtocolConfig> mLocalMRPConfig;
-    std::optional<bool> mTcpSupported;
+    TCPModeAdvertise mTcpSupportModes  = TCPModeAdvertise::kNone;
```

Comment:
the std::optional was there to explicity have support for "no T option" vs saying "T=0" for example. 

This update makes it so that "T=0" is never exposed anymore. Is this the correct path to lose that capability?